src/main/java/blusunrize/immersiveengineering/common/data/Recipes.java

```diff
@@ -659,73 +1442,81 @@ protected void registerRecipes(@Nonnull Consumer<IFinishedRecipe> out)
+			.addCriterion("has_"+toPath(IEBlocks.Connectors.getEnergyConnector(WireType.LV_CATEGORY, false)), hasItem(IEBlocks.Connectors.getEnergyConnector(WireType.LV_CATEGORY, false)))
 			.build(out);
-
-		CustomRecipeBuilder.func_218656_a(RecipeSerializers.SPEEDLOADER_LOAD.get())
-			.build(out, ImmersiveEngineering.MODID+":speedloader_load");
-		CustomRecipeBuilder.func_218656_a(RecipeSerializers.FLARE_BULLET_COLOR.get())
-			.build(out, ImmersiveEngineering.MODID+":flare_bullet_color");
-		CustomRecipeBuilder.func_218656_a(RecipeSerializers.POTION_BULLET_FILL.get())
-			.build(out, ImmersiveEngineering.MODID+":potion_bullet_fill");
-		CustomRecipeBuilder.func_218656_a(RecipeSerializers.JERRYCAN_REFILL.get())
-			.build(out, ImmersiveEngineering.MODID+":jerrycan_refill");
-
-		//NYI
-//		ShapedRecipeBuilder.shapedRecipe(IEItems.Misc.steelArmor[0]).patternLine("i i").patternLine("i i").key('i', IETags.getTagsFor(EnumMetals.STEEL).ingot).addCriterion("has_steel_ingot", this.hasItem(IETags.getTagsFor(EnumMetals.STEEL).ingot)).build(out);
-//		ShapedRecipeBuilder.shapedRecipe(IEItems.Misc.steelArmor[1]).patternLine("iii").patternLine("i i").patternLine("i i").key('i', IETags.getTagsFor(EnumMetals.STEEL).ingot).addCriterion("has_steel_ingot", this.hasItem(IETags.getTagsFor(EnumMetals.STEEL).ingot)).build(out);
-//		ShapedRecipeBuilder.shapedRecipe(IEItems.Misc.steelArmor[2]).patternLine("i i").patternLine("iii").patternLine("iii").key('i', IETags.getTagsFor(EnumMetals.STEEL).ingot).addCriterion("has_steel_ingot", this.hasItem(IETags.getTagsFor(EnumMetals.STEEL).ingot)).build(out);
-//		ShapedRecipeBuilder.shapedRecipe(IEItems.Misc.steelArmor[3]).patternLine("iii").patternLine("i i").key('i', IETags.getTagsFor(EnumMetals.STEEL).ingot).addCriterion("has_steel_ingot", this.hasItem(IETags.getTagsFor(EnumMetals.STEEL).ingot)).build(out);
+		// TODO Uncomment below when maintenanceKit is implemented
+//		ShapedRecipeBuilder.shapedRecipe(IEItems.Misc.maintenanceKit)
+//			.patternLine("rc ")
+//			.patternLine("fff")
+//			.key('c', Ingredient.fromItems(Items.SHEARS, Tools.wirecutter))
+//			.key('r', IETags.ironRod)
+//			.key('f', IETags.fabricHemp)
+//			.addCriterion("has_"+toPath(Tools.wirecutter)), hasItem(Tools.wirecutter))
+//			.build(out);
+		// TODO Uncomment below when shield is implemented
+//		ShapedRecipeBuilder.shapedRecipe(IEItems.Misc.shield)
+//			.patternLine("sws")
+//			.patternLine("scs")
+//			.patternLine("sws")
+//			.key('s', IETags.getTagsFor(EnumMetals.STEEL).plate)
+//			.key('w', IETags.getItemTag(IETags.treatedWood))
+//			.key('c', Items.SHIELD)
+//			.addCriterion("has_shield", hasItem(Items.SHIELD))
+//			.build(out);
 	}
 
-	//TODO use tags
-	private void add3x3Conversion(IItemProvider big, IItemProvider small, Consumer<IFinishedRecipe> out)
+	// Experimental?
+	private void add3x3Conversion(IItemProvider bigItem, Tag<?> bigTag, IItemProvider smallItem, Tag<?> smallTag, Consumer<IFinishedRecipe> out)
```

Comment:
Why are these `Tag<?>` instead of `Tag<Item>`?

Reply:
Because of `The method add3x3Conversion(IItemProvider, Tag<Item>, IItemProvider, Tag<Item>, Consumer<IFinishedRecipe>) in the type Recipes is not applicable for the arguments (Block, Tag<Block>, Item, Tag<Item>, Consumer<IFinishedRecipe>)`

Thats why i did `Tag<?>` instead to have it be more general, and it works so without trouble so far.
And now after playing around with it, only the first tag needs to be `Tag<?>` because that gets used for both Block and Item.

Reply:
I'm not sure what exactly casting a `Tag<Block>` to `Tag<Item>` will do, it will probably just crash when trying to use any `Item`-related method. You can use `IETags#getItemTag` to get the item tag for any IE block tag.